### ISSUE TITLE
build(ci): improve edge cases CI

### ISSUE THREAD

**Issue by ThomasHepworth** [RobinL/uk_address_matcher#214]
At some point, it would be good to formally review our edge cases CI step so we can both better review the performance of our model(s) and as a way of more efficiently iterating on the model itself using gradient descent or other tools.

As we build and iterate, it would be good to identify and record some of the stranger edge cases that we've encountered, to help with building this out. It's also important that we also look to include more standard examples so we don't overfit to edge cases.

See:
- https://github.com/moj-analytical-services/uk_address_matcher/blob/main/.github/workflows/edge_cases_github_comment.yml
